File: UsingRoomORM/app/src/main/java/nisrulz/github/sample/usingroomorm/PersonDao.java
package nisrulz.github.sample.usingroomorm;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import java.util.List;

@Dao
public interface PersonDao {

  // Adds a person to the database
  @Insert
  void insertAll(Person... people);

  // Removes a person from the database
  @Delete
  void delete(Person person);

  // Gets all people in the database
  @Query("SELECT * FROM person")
  List<Person> getAllPeople();

  // Gets all people in the database with a address
  @Query("SELECT * FROM person WHERE address LIKE :address")
  List<Person> getAllPeopleWithAddress(String address);
}

File: UsingRoomORM/app/src/main/java/github/nisrulz/example/usingroomorm/PersonDao.kt
package github.nisrulz.example.usingroomorm

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface PersonDao {
    // Adds a person to the database
    @Insert
    fun insertAll(vararg people: Person)

    // Removes a person from the database
    @Delete
    fun delete(person: Person)

    // Gets all people in the database
    @get:Query("SELECT * FROM person")
    val allPeople: List<Person?>?

    // Gets all people in the database with a address
    @Query("SELECT * FROM person WHERE address LIKE :address")
    fun getAllPeopleWithAddress(address: String): List<Person?>?
}